modules/core/map/js/farmOS.map.behaviors.input.js

```diff
@@ -1,11 +1,56 @@
 (function () {
   farmOS.map.behaviors.input = {
     attach: function (instance) {
+
+      // Get the data input form element.
+      var input = instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]')
+
+      // When features change in the edit layer, write WKT to the input form element.
       instance.editAttached && instance.editAttached.then(() => {
         instance.edit.wktOn('featurechange', function(wkt) {
-          instance.map.getTargetElement().parentElement.querySelector('[data-map-geometry-field]').value = wkt;
+          input.value = wkt;
         });
```

Comment:
Just curious how we avoid an infinite loop if the input updates the features and the features update the input?

Reply:
I wondered this too, but apparently this does not trigger the `featurechange` event, and so it doesn't cause the input field to get updated again.

Reply:
Could check to see if the input already has the same value that it would be replacing. If it is the same, don't update the value

Reply:
@paul121 That make sense. Will it be challenging to compare them, though? We have the new WKT, and we have the old map layer source features. I don't think we have a public method of converting those features to WKT, so we can't compare WKT to WKT. Maybe we can convert the new WKT to features and compare OL features to OL features? I wonder if that will get complicated if there are multiple features and they end up in different orders?

Reply:
Hmm I was just thinking to do this check in the wkt event listener right before replacing the input value, when we do have the WKT string?

```js
if (input.value != wkt) {
  input.value = wkt;
}
```

Reply:
But maybe the browser already handled this situation and won't fire a change event if the value doesn't change?

Reply:
Oooh I see. I thought you meant in the new `input` event listener... Yea that makes perfect sense. I'll push a fixup to add that...

Reply:
https://github.com/farmOS/farmOS/pull/670/commits/16642b62cf25fb34f4c6c09becd5965cd4cf520f